File: server/src/main/java/rs/ac/singidunum/novisad/server/model/RealizacijaPredmeta/TerminNastave.java
package rs.ac.singidunum.novisad.server.model.RealizacijaPredmeta;

import jakarta.persistence.*;
import lombok.*;
import rs.ac.singidunum.novisad.server.model.predmet.Ishod;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
public class TerminNastave {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDateTime vremePocetka;
    private LocalDateTime vremeZavrsetka;
    @OneToOne
    private Ishod ishod;
    @ManyToOne
    private TipNastave tipNastave;
    @ManyToOne
    private RealizacijaPredmeta realizacijaPredmeta;
}

File: client/src/app/model/terminNastave.ts
import {Ishod} from "./ishod";
import {TipNastave} from "./tipNastave";
import {NastavniMaterijal} from "./nastavniMaterijal";

export interface TerminNastave {
  id: number | null,
  vremePocetka: Date,
  vremeZavrsetka: Date,
  ishod: Ishod | null,
  tipNastave: TipNastave | null,
  nastavniMaterijal: NastavniMaterijal | null
  event?: any
}
